### ISSUE TITLE
Accelerate gradient descent through known acceleration techniques

### ISSUE THREAD

**Issue by rgeronimi** [catboost/catboost#277]
Problem: gradient descent could be accelerated (less trees, better generalization) through gradient descent acceleration techniques
catboost version: all
Operating System: all
CPU: all
# GPU: all

See https://github.com/Microsoft/LightGBM/issues/1257
and
https://arxiv.org/pdf/1803.02042.pdf

The idea is basically to apply gradient descent acceleration techniques (e.g., Nesterov, Adagrad, Adam). The experimental results look promising.

**Comment by Noxoomo** [comment]
Hi!

Thank you for suggestion, we will look at it.

But, AFAIK, some of our researchers already experimented with such techniques several years ago and there were no profit (but on specific for Yandex datasets)

**Comment by Noxoomo** [comment]
Plus for problems for most problems we solve there are "free" second-order  derivatives
